### PR TITLE
[DOC] Updated README.md's redirection link to Kaizen documentation

### PR DESCRIPTION
**Pull request by SwapnilChand**
# Update README for GitHub App Setup Instructions

## Overview
This pull request aims to enhance the clarity and conciseness of the README documentation related to the setup of the GitHub app for code review functionalities.

## Changes
- Key Changes: 
  - Consolidated instructions regarding the necessity of the GitHub app for code review and clarified that other functionalities do not require it.
  - Updated the link reference for the self-hosting guide to improve user navigation.

- New Features: 
  - None identified in this patch.

- Refactoring: 
  - Improved wording for better readability and comprehension in the GitHub App Setup section.

<details>

> ✨ Generated with love by [Kaizen](https://cloudcode.ai) ❤️

<details>
<summary>Original Description</summary>

</details>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,10 +118,9 @@ Kaizen takes a two-pronged approach to help you find and squash bugs, both befor
 
 ### 🔧 GitHub App Setup
 
-You only need to install the GitHub app for code review. Other functionalities don't need a GitHub app.
-For PR review and description updates:
+You only need to install the GitHub app for code review (PR review and description updates). Other functionalities don't need a GitHub app.
 
-1. Follow our [GitHub App Setup Guide](https://cloudcode.ai/kaizen/docs/self_hosting_guide/setup_github_app)
+1. For the self-hosting guide, visit the [Kaizen Documentation](https://cloudcode.ai/kaizen/docs) and navigate to the "Self Hosting Guide" section.
```
Comment (sauravpanda) [inline README.md:123]: cant we pass that in the link itself?

**Review by sauravpanda — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (sauravpanda) [reply README.md:123]: Here is the link: https://cloudcode.ai/kaizen/docs/self_hosting_guide/setup_github_app

Comment (sauravpanda) [reply README.md:123]: hmm seems weird, we cant load the link when it has more specific path, need to work on it to find a fix

Comment (sauravpanda) [reply README.md:123]: For now lets use what we have here, approving it

Comment (SwapnilChand) [reply README.md:123]: https://www.perplexity.ai/search/why-is-it-that-i-cant-go-to-th-uHeM6WZrRkOlSkbWpRDMEA#0
Not sure why though it seems to be how github pages hosting is. 
The previous edit (https://github.com/Cloud-Code-AI/kaizen/pull/498) was this link itself.

**Pull request merged**